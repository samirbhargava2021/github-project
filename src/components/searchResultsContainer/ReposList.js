import React from "react";
import HorizontalCard from "../horizontalCard/HorizontalCard";
import "./ReposList.scss";

const ReposList = ({ list }) => {
  return (
    <div className="repo-section">
      {
        <>
          {list.map((item, index) => (
            <>
              <HorizontalCard key={index} repo={item.node} />
              <hr />
            </>
          ))}
        </>
      }
    </div>
  );
};

export default ReposList;
