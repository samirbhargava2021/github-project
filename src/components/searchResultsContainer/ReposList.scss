.repo-section {
  margin: 0 auto;
  margin: 0 5% 2% 5%;
  .repo-cards {
    display: flex;
    justify-content: center;
    align-items: center;
    padding-bottom: 24px;
  }
}

hr {
  border: none;
  height: 20px;
  height: 50px;
  margin-top: 0;
  border-bottom: 1px solid #fff9f5;
  box-shadow: 0 20px 20px -20px #d7d2d2;
  margin: -50px auto 10px;
}