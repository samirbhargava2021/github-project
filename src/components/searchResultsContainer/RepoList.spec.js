import { shallow } from "enzyme";
import { render, fireEvent  } from "@testing-library/react";
import ReposList from "./ReposList";

describe("ReposList Component", () => {
    const Props = {
        list: [
            {
                node :{
                    languages: {
                        nodes: [
                          {
                            name: "JavaScript",
                          },
                          {
                            name: "CSS",
                          }
                        ],
                      },
                      name: "react",
                      url: "https://github.com/typescript-cheatsheets/react",
                      description: "Cheatsheets for experienced React developers getting started with TypeScript",
                      stargazerCount: 21954,
                      updatedAt: "2021-02-18T02:44:13Z",
                      openGraphImageUrl: "https://repository-images.githubusercontent.com/135786093/b56ad700-6d50-11e9-9775-9f64c1e1414e",
                    owner: {
                        avatarUrl: 'https://avatars.githubusercontent.com/u/50188264?v=4',
                        login: 'typescript-cheatsheets'
                        }
                }
            },
        ]
    }
  it("should render ReposList Component", () => {
    const component = shallow(<ReposList list={Props.list}/>);
    expect(component.getElements()).toMatchSnapshot();
  });
  it("should have className repo-section to load the ReposList component with proper styling", () => {
    const { container } = render(<ReposList list={Props.list}/>);
    expect(container.firstChild.classList.contains('repo-section')).toBe(true)
  });
});
