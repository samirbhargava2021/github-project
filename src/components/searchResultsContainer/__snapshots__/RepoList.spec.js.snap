// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReposList Component should render ReposList Component 1`] = `
Array [
  <div
    className="repo-section"
  >
    <React.Fragment>
      <React.Fragment>
        <HorizontalCard
          repo={
            Object {
              "description": "Cheatsheets for experienced React developers getting started with TypeScript",
              "languages": Object {
                "nodes": Array [
                  Object {
                    "name": "JavaScript",
                  },
                  Object {
                    "name": "CSS",
                  },
                ],
              },
              "name": "react",
              "openGraphImageUrl": "https://repository-images.githubusercontent.com/135786093/b56ad700-6d50-11e9-9775-9f64c1e1414e",
              "owner": Object {
                "avatarUrl": "https://avatars.githubusercontent.com/u/50188264?v=4",
                "login": "typescript-cheatsheets",
              },
              "stargazerCount": 21954,
              "updatedAt": "2021-02-18T02:44:13Z",
              "url": "https://github.com/typescript-cheatsheets/react",
            }
          }
        />
        <hr />
      </React.Fragment>
    </React.Fragment>
  </div>,
]
`;
